client/packages/common/src/ui/components/inputs/Filters/DateFilter.tsx

```diff
@@ -86,22 +85,40 @@ const getDateFromUrl = (query: string, range: RangeOption | undefined) => {
   return DateUtils.getDateOrNull(value);
 };
 
-const getRangeBoundary = (
+export const getMinMaxDates = (
   query: string | RangeObject<string>,
   range: RangeOption | undefined,
-  limit: Date | string | undefined
+  min: Date | string | undefined,
+  max: Date | string | undefined
 ) => {
-  const limitDate = DateUtils.getNaiveDate(limit);
-  if (typeof query !== 'object' || !range) return limitDate || undefined;
+  const minDate = DateUtils.getDateOrNull(min);
+  const maxDate = DateUtils.getDateOrNull(max);
+
+  if (typeof query !== 'object' || !range) {
+    return {
+      minDate: minDate || undefined,
+      maxDate: maxDate || undefined,
+    };
+  }
+
   const { from, to } = query as RangeObject<string>;
 
-  if (range === 'from')
-    return to
-      ? (DateUtils.minDate(DateUtils.getNaiveDate(to), limitDate) ?? undefined)
-      : (limitDate ?? undefined);
-  else
-    return from
-      ? (DateUtils.maxDate(DateUtils.getNaiveDate(from), limitDate) ??
-          undefined)
-      : (limitDate ?? undefined);
+  if (range === 'from') {
+    const toDate = DateUtils.getDateOrNull(to);
+    return {
+      minDate: minDate || undefined,
```

Comment:
Do you reckon we restrict anywhere, minData for 'from field' ?

Reply:
Not sure what you mean? Only would be restricted if you defined a min in the date filter definition?

Reply:
Oki, i think we need to review this whole thing at some point lol

Reply:
agree... along with other date issues 🥲